library/src/main/java/com/vinted/coper/CoperBuilder.kt

```diff
@@ -1,37 +1,72 @@
 package com.vinted.coper
 
+import androidx.fragment.app.FragmentActivity
 import androidx.fragment.app.FragmentManager
 
 /**
  * [Coper] instance builder.
  * Use case:
  * ```
  * val coper = CoperBuilder()
- *      .setFragmentManager(fragmentManager)
+ *      .setFragmentActivity(activity)
  *      .build()
  * ```
  * If [fragmentManager] will be null on [build], then it will throw [IllegalArgumentException].
  */
 class CoperBuilder {
 
+    private var activity: FragmentActivity? = null
+
+    private var resultFragmentCreationTimeout: Long? = null
+
     private var fragmentManager: FragmentManager? = null
 
+    /**
+     * Set fragment activity for permission handling.
+     */
+    fun setFragmentActivity(fragmentActivity: FragmentActivity): CoperBuilder {
+        this.activity = fragmentActivity
+        return this
+    }
+
     /**
      * Set fragment manager for permission handling.
      */
+    @Deprecated(
+        message = "Please use setFragmentActivity as it will provide safer usage",
+        replaceWith = ReplaceWith("setFragmentActivity(activity)")
+    )
     fun setFragmentManager(fragmentManager: FragmentManager): CoperBuilder {
         this.fragmentManager = fragmentManager
         return this
     }
 
+    /**
+     * Set timeout for result fragment preparation.
+     * Default -> 1s
+     */
+    fun setResultFragmentPreparationTimeout(timeout: Long): CoperBuilder {
```

Comment:
When this fragment preparation could timeout?

Reply:
When it was not possible to add to fragment manager and commit to fragment manager was delayed longer than it should be, as it is not instantly it should help to prevent not ending function calls

Reply:
How could it happen? For me, it sounds like race-condition.

Reply:
Probably in real conditions it should not happen, but as it is callback on commit to fragment manager or callback when activity or fragment will be ready to work, when they will reach onCreate and it could be delayed for example with activity not ready or dispatcher being not able to change from IO to Main, for some reason, I added timeout. What do you think is it good enough to have it, or client should add these callbacks by themselfes?

Reply:
As a user of this library, I expected no hidden timeouts. I always could put some on the consumer side and I expect the library to proper cancel the request

Reply:
Removed this one, good points, client will be able to add it itself